src/main/java/org/folio/linked/data/integration/kafka/sender/search/KafkaSearchSenderImpl.java

```diff
@@ -0,0 +1,116 @@
+package org.folio.linked.data.integration.kafka.sender.search;
+
+import static java.lang.Long.parseLong;
+import static org.folio.linked.data.util.BibframeUtils.isSameResource;
+import static org.folio.linked.data.util.Constants.FOLIO_PROFILE;
+import static org.folio.linked.data.util.Constants.SEARCH_RESOURCE_NAME;
+import static org.folio.search.domain.dto.SearchIndexEventType.CREATE;
+import static org.folio.search.domain.dto.SearchIndexEventType.DELETE;
+import static org.folio.search.domain.dto.SearchIndexEventType.UPDATE;
+
+import java.util.List;
+import java.util.UUID;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.extern.log4j.Log4j2;
+import org.folio.linked.data.mapper.kafka.KafkaSearchMessageMapper;
+import org.folio.linked.data.model.entity.Resource;
+import org.folio.linked.data.model.entity.event.ResourceIndexedEvent;
+import org.folio.search.domain.dto.BibframeIndex;
+import org.folio.search.domain.dto.SearchIndexEvent;
+import org.folio.spring.tools.kafka.FolioMessageProducer;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+
+@Log4j2
+@Service
+@Profile(FOLIO_PROFILE)
+@RequiredArgsConstructor
+public class KafkaSearchSenderImpl implements KafkaSearchSender {
+
+  private final ApplicationEventPublisher eventPublisher;
+  private final KafkaSearchMessageMapper kafkaSearchMessageMapper;
+  private final FolioMessageProducer<SearchIndexEvent> searchIndexEventMessageProducer;
+
+  @SneakyThrows
+  @Override
+  public void sendSingleResourceCreated(Resource resource) {
+    kafkaSearchMessageMapper.toIndex(resource, CREATE).ifPresent(bibframeIndex -> sendCreate(bibframeIndex, true));
+  }
+
+  @Override
+  public boolean sendMultipleResourceCreated(Resource resource) {
+    return kafkaSearchMessageMapper.toIndex(resource, CREATE)
+      .map(bibframeIndex -> {
+        sendCreate(bibframeIndex, false);
+        return true;
+      })
+      .isPresent();
+  }
+
+  @SneakyThrows
+  private void sendCreate(BibframeIndex bibframeIndex, boolean publishIndexEvent) {
```

Comment:
minor
using `if boolean parameter` isn't good practice
suggest to make separate methods
```java
private void sendCreateWithPublishingIndex(BibframeIndex bibframeIndex) {
    sendCreate(bibframeIndex)
   eventPublisher.publishEvent(new ResourceIndexedEvent(parseLong(bibframeIndex.getId())));
}

private void sendCreate(BibframeIndex bibframeIndex) {
    var message = new SearchIndexEvent()
      .id(UUID.randomUUID().toString())
      .type(CREATE)
      .resourceName(SEARCH_RESOURCE_NAME)
      ._new(bibframeIndex);
    searchIndexEventMessageProducer.sendMessages(List.of(message));
}

```

Reply:
agree, and we could make it even more optimal just separating methods. Changed, thx!